src/groundStation/commandParser.py

```diff
@@ -108,10 +108,29 @@ def parseReturnValue(self, src, dst, dport, data, length):
 
         print(length)
         returns = subservice['inoutInfo']['returns']
+        args = subservice['inoutInfo']['args']
         for retVal in returns:
-            outputObj[retVal] = np.frombuffer(
-                data, dtype=returns[retVal], count=1, offset=idx)[0]
-            idx += np.dtype(returns[retVal]).itemsize
+            if returns[retVal] == 'var' and dport == 9 and subservice['subPort'] < 4:
+                #Variable size config return
+                if outputObj['type'] == 0:
+                    config_type = np.dtype('<u1')
+                elif outputObj['type'] == 1:
+                    config_type = np.dtype('<i1')
+                elif outputObj['type'] == 2:
+                    config_type = np.dtype('<u2')
+                elif outputObj['type'] == 4:
+                    config_type = np.dtype('<u4')
+                elif outputObj['type'] == 9:
+                    config_type = np.dtype('<S16') #Empty means all zero or Use <V16
```

Comment:
Any comments on how to improve this? It basically looks at the byte before and decides what data type will be the rest of the packet.

Reply:
yes let's make a dictionary that can hash all these types so we can just do something more like
```
dtype = returns[retval] if returns[retval] != 'var' else varTypes[outputObj['type']]
outputObj[retVal] = np.frombuffer(
                    data, dtype=dtype, count=1, offset=idx)[0]
...
```

Reply:
It seems like we could probably not check the port and subservices and if we ever need this again we can conform to this design

Reply:
Oh, I didn't see this comment before.
Sounds good. Yeah, I just wanted to make it work that day. That would be a good implementation. Thanks.
Ok, I'll remove the check for port/subport.
Please look at NA-EPS-ICD-R6 sections 16.1 and 3.5 to see what's happening. It's for EPS so there's no obc code.

Reply:
okay sounds good